File: app/src/main/java/com/akshaj02/shopnsplit/Split.java
package com.akshaj02.shopnsplit;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;


public class Split extends AppCompatActivity {

    private RecyclerView recyclerView;
    private List<SplitModel> mList;
    private SplitAdapter adapter;
    LinearLayoutManager layoutManager;
    private int counter = 0;

    Button mHomepage;
    Button mSplit;

    String date;

    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_split);

        //if the intent is null, then do this
        Intent intent = getIntent();
        String expense = intent.getStringExtra("description");
        String moneyOwed = intent.getStringExtra("money");
        date = intent.getStringExtra("date");


        //convert moneyOwed to a double


//        String contactList = intent.getStringExtra("contacts");
//        int count = 4;
//        int count = Integer.parseInt(contactList.split(",").length + "");
//        //replace every character with " " if it is between | and ,
//        String contactList1 = contactList.replaceAll("\\|[^,]*\\,", " ");
//        //remove | from the string
//        String contactList2 = contactList1.replaceAll("\\|", "");
//        //remove the space before comma
//        String contactList3 = contactList2.replaceAll(" ,", ",");
//        //remove the comma at the end
//        String contactList4 = contactList3.replaceAll(",$", "");
        String amountLent = "15.00";
        //Split the money owed by the number of contacts
//        double money = Double.parseDouble(moneyOwed);
//        double moneySplit = money/count;
//        double total = moneySplit * (count-1);
//        String amountLent = Double.toString(total);

        initData(expense, date, moneyOwed, amountLent);
        initRecyclerView();


        //mSplit action
        mSplit = findViewById(R.id.split_button);
        mSplit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                counter++;
                Intent intent = new Intent(Split.this, AddExpense.class);
                startActivity(intent);
            }
        });

    }

    private void initRecyclerView() {
        recyclerView = findViewById(R.id.splitRecycler);
        layoutManager = new LinearLayoutManager(this);
        layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        recyclerView.setLayoutManager(layoutManager);
        adapter = new SplitAdapter(mList);
        recyclerView.setAdapter(adapter);
        adapter.notifyDataSetChanged();

    }
    private void initData(String expense, String date, String moneyOwed, String amountLent) {
        mList = new ArrayList<>();
        //Hardcoded data, need to get data from AddExpense.java
        mList.add(new SplitModel(expense, date, "200", "250"));
    }
}

File: app/src/main/java/com/akshaj02/shopnsplit/Split.kt
package com.akshaj02.shopnsplit;

import android.annotation.SuppressLint
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity;
import org.w3c.dom.Text


//class Split : AppCompatActivity() {
//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_split)
//
//        val addexpense = findViewById<Button>(R.id.split)
//        addexpense.setOnClickListener {
//            val intent = Intent(this, AddExpense::class.java)
//            startActivity(intent)
//        }
//
//    }
//
//}

class Split : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_split)

        val amountPrint = findViewById<TextView>(R.id.money)
        val description = findViewById<TextView>(R.id.desc)
        val contactsss = findViewById<TextView>(R.id.people)
        //get intent from previous activity
        val intent = getIntent()
        val description1 = intent.getStringExtra("description")
        val amount1 = intent.getStringExtra("money")
        val contacts = intent.getStringExtra("contacts")


        //Get number of contacts and split amount
        val numContacts = contacts?.split(",")?.size
        val splitAmount = amount1?.toDouble()?.div(numContacts?.toDouble()!!)
        //Multiple the amount by size - 1 to get the amount to be paid by the person who added the expense
        val amountToBePaid = splitAmount?.toDouble()?.times(numContacts?.minus(1)!!)
        //Print the amount to be paid by the person who added the expense
        amountPrint.text = amountToBePaid.toString()
        //Print the description of the expense

        //replace every character with " " if it is between | and ,
        val contacts1 = contacts?.replace(Regex("(?<=\\|)[^,]*(?=,)"), " ")
        //remove | from the string
        val contacts2 = contacts1?.replace("|", "")
        //remove the space before comma
        val contacts3 = contacts2?.replace("  ,", ", ")
        //remove the comma at the end
        val contacts4 = contacts3?.replace(",$", "")
        //add a $ before amount
        val amount2 = "$" + amount1


        //set the textviews to the values
        description.text = description1
        contactsss.text = contacts4


        val addexpense = findViewById<Button>(R.id.split)
        addexpense.setOnClickListener {
            val intent = Intent(this, AddExpense::class.java)
            startActivity(intent)
        }



    }

}